pkg/operator/operator.go

```diff
@@ -77,43 +82,34 @@ func (c OpenShiftControllerManagerOperator) sync() error {
 	if err != nil {
 		return err
 	}
+	// manage status
+	originalOperatorConfig := operatorConfig.DeepCopy()
+	reasonString := ""
+	messageString := ""
 	switch operatorConfig.Spec.ManagementState {
+	case operatorapiv1.Removed:
+		fallthrough
+		// we equally do not allow removed/unmanaged
 	case operatorapiv1.Unmanaged:
-		// manage status
-		originalOperatorConfig := operatorConfig.DeepCopy()
-		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorapiv1.OperatorCondition{
-			Type:    operatorapiv1.OperatorStatusTypeAvailable,
-			Status:  operatorapiv1.ConditionUnknown,
-			Reason:  "Unmanaged",
-			Message: "the controller manager is in an unmanaged state, therefore its availability is unknown.",
-		})
-		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorapiv1.OperatorCondition{
-			Type:    operatorapiv1.OperatorStatusTypeProgressing,
-			Status:  operatorapiv1.ConditionFalse,
-			Reason:  "Unmanaged",
-			Message: "the controller manager is in an unmanaged state, therefore no changes are being applied.",
-		})
-		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorapiv1.OperatorCondition{
-			Type:    operatorapiv1.OperatorStatusTypeDegraded,
-			Status:  operatorapiv1.ConditionFalse,
-			Reason:  "Unmanaged",
-			Message: "the controller manager is in an unmanaged state, therefore no operator actions are failing.",
-		})
-
-		if !equality.Semantic.DeepEqual(operatorConfig.Status, originalOperatorConfig.Status) {
-			if _, err := c.operatorConfigClient.OpenShiftControllerManagers().UpdateStatus(operatorConfig); err != nil {
-				return err
-			}
-		}
-		return nil
+		reasonString = fmt.Sprintf("%sUnsupported", string(operatorConfig.Spec.ManagementState))
+		messageString = fmt.Sprintf("the controller manager spec was set to %s state, but that is unsupported, and has no effect on this condition", string(operatorConfig.Spec.ManagementState))
 
-	case operatorapiv1.Removed:
-		// TODO probably need to watch until the NS is really gone
-		if err := c.kubeClient.CoreV1().Namespaces().Delete(util.TargetNamespace, nil); err != nil && !apierrors.IsNotFound(err) {
+		// as we are ignoring / not supporting unmanaged/removed, we still process any other inputs to the sync/reconciliation
+		// unlike what we used to do
+	case operatorapiv1.Managed:
+		// we want to empty out the reason/message string if transitioning from the other phases so the default setting above is good
+	}
+
+	for _, condition := range operatorConfig.Status.Conditions {
+		// do not change the current status as part of noting that unmanaged/removed are not supported
+		condition.Reason = reasonString
+		condition.Message = messageString
```

Comment:
This should not clear the previous message/reason.

if i set the management state to "unmanaged", and it's being ignored, i should not lose the pre-existing reasons/messages for whatever condition my operator is in (or enters in the future).

Reply:
that is not what will happen here, nor what I saw in my testing

Though we are "ignoring" unmanaged/removed, it stays set in the Spec.  So we will reset the reason/message with the values we want for unmanaged/removed each time ... on each event.

The code below simply will not issue another updatestatus if the fields did not change since the last pass.

If that doesn't address your concern, I'm either not understanding your concern or missing how your concern enters the mix here.

Reply:
My reading of the code is that the behavior will be this:

starting state:  managed
available=true, reason=Normal, message=everything is great

i change it to state: unmanaged, the condition reasons/message will change to:
available=true, reason="unmanaged is being ignored", message="unmanaged is being ignored"

is that right?  If so, that is what i think is incorrect behavior.  Yes i need to know my attempt to set the management state was ignored, but i should also continue to be told the actual reasons/message for the actual state the operator is in.

Reply:
Reason/Message are blank when managed and available=true and everything is good.

And in fact, when examining `syncOpenShiftControllerManager_v311_00_to_latest`, the only time Reason/Message are set is when there is an error.

And `syncOpenShiftControllerManager_v311_00_to_latest` will override the "unmanaged is being ignored" message/reason we set here if in fact it finds something wrong 

So this really is just a replacement of blank reason/message, not a replacement of non-blank reason/message

Reply:
ok